third_party/nvfuser/csrc/fusion_segmenter.cpp

```diff
@@ -892,11 +891,31 @@ std::vector<SegmentedEdge*> SegmentedFusion::castInputOutputToLowerPrecision(
     // in the merged groups to test if they can indeed be merged.
     if (std::find(groups_to_merge.begin(), groups_to_merge.end(), edge->to) !=
         groups_to_merge.end()) {
-      for (auto edge_val_use_expr : edge_tv->uses()) {
-        if (!merged_group_exprs.count(edge_val_use_expr)) {
-          uses_to_keep.emplace(edge_val_use_expr);
-        }
-      }
+      uses_to_modify.erase(
+          std::remove_if(
+              uses_to_modify.begin(),
+              uses_to_modify.end(),
+              [&](Expr* edge_val_use_expr) {
+                return !merged_group_exprs.count(edge_val_use_expr);
+              }),
+          uses_to_modify.end());
```

Comment:
Did you mean the following?
```C++
std::remove_if(
    uses_to_modify.begin(),
    uses_to_modify.end(),
    [&](Expr* edge_val_use_expr) {
      return !merged_group_exprs.count(edge_val_use_expr);
    }),
```
I don't understand why you uses both `erase` and `remove_if`.

Reply:
I thought that's necessary to actually erase. https://en.cppreference.com/w/cpp/algorithm/remove 

Does `remove_if` actually change the `end` iterator? I thought that's done by `erase`.

Reply:
```
#include <vector>
#include <iostream>
#include <algorithm>

void test1(std::vector<int> x) {
  std::remove_if(x.begin(), x.end(),
                 [](auto i) {
                   return i % 2 == 0;
                 });

  for (auto i: x) {
    std::cout << "test1: " << i << std::endl;
  }
}

void test2(std::vector<int> x) {
  x.erase(std::remove_if(x.begin(), x.end(),
                         [](auto i) {
                           return i % 2 == 0;
                         }), x.end());

  for (auto i: x) {
    std::cout << "test2: " << i << std::endl;
  }
}

int main() {
  std::vector<int> x({1, 2, 3, 4});
  test1(x);
  test2(x);
  return 0;
}
```
Output:
```
test1: 1
test1: 3
test1: 3
test1: 4
test2: 1
test2: 3
```

So, looks like without `erase` the end iterator is not moved.

Reply:
Oh, I didn't know that. Thanks for explaining!

Reply:
NP. There's `std::erase` in C++20, which would be much simpler.